### ISSUE TITLE
Add support lbry.tv / odysee.com

### ISSUE THREAD

**Issue by KonoromiHimaries** [TeamNewPipe/NewPipeExtractor#434]

**Comment by cauerego** [comment]
dear reader, you might want to check the linked topic there for now context on this feature request!

https://github.com/TeamNewPipe/NewPipe/issues/4733

disclaimer: i wrote it. 😁

**Comment by Neurognostic** [comment]
No extractor is needed to parse lbry.tv or odysee.com, instead videos can be read straight from the LBRY network.

[`lbt`](https://gitlab.com/gardenappl/lbt) is a feature complete shell script for interacting with the LBRY network simply by using `curl` and the [LBRY SDK](https://github.com/lbryio/lbry-sdk/) `lbrynet`.

**Comment by The-Quantum-Alpha** [comment]
LBRY in Newpipe when?

**Comment by Yur7y** [comment]
Any updates on the status of this?

**Comment by vachan-maker** [comment]
Hello,
This is something I am waiting for.
piped.kavin.rocks already supports lbry streaming.

**Comment by BloodRaven0** [comment]
Yes please. Also don't pull that bitchute bullshit on us this time please!

**Comment by mcdviii** [comment]
If this is written--will the pull request be accepted, or will it be slapped down on account of some kind of ideology? I'd like to foster competition to YT & do not want to use the LBRY app because it's not as efficient as Newpipe.

**Comment by strider72** [comment]
> If this is written--will the pull request be accepted, or will it be slapped down on account of some kind of ideology?

NewPipe would slap it down. Take a look at a fork called BraveNewPipe. I believe they would happily accept your contribution. BNP is NewPipe with additional sources, and no political ideology

**Comment by The-Quantum-Alpha** [comment]
Well, I know who I will support by tipping money!
bravenewpipe!

**Comment by mrusme** [comment]
Funded this issue via gitpay.me: https://gitpay.me/#/task/608

Anyone who's interested in this, feel free to chip in. Maybe we can get it done!

**Comment by DraconicNEO** [comment]
Is this going to be added or are you going to close the issue? Naturally people are hesitant to make a pull request because of how quickly the other one was shut down.

**Comment by opusforlife2** [comment]
> I hope you guys are aware of the difference between closing a feature request that took little effort and rejecting a pull request that took a lot of effort.

@DraconicNEO Doesn't matter. If you read the contribution guidelines, you will see that we strongly advise contributors to get in touch with us _before_ doing anything major, _precisely_ to avoid wasted effort. If a contributor ignores that, it is on them.

BTW, there was no such pull request. There have only been issues asking for support. You might be thinking of another project or repo.

**Comment by mongolsteppe** [comment]
This fork supports bitchute and rumble already (not odysee unfortunately)
https://github.com/bravenewpipe/NewPipe

I suggest switching over, since NewPipe devs are unlikely to implement support for these sites due to their political views

**Comment by brunocalmon** [comment]
Honest question:

What political reasons makes them be against LBRY/Odysee though?

It is just an decentralized alternative to YT ran by individuals servers connected by the BitTorrent Protocol, the Blockchain is simply for indexing trackers.

No right/left wing speech are promoted by them, and they have basically the same rules of YouTube.

=========

My guessing: 

Maybe it's not political though, but tech-idealism? 
Since they have PeerTube, that is basically the same of Odysee but part of Fediverse my guessing would be that the developers don't actually like blockchain stuff and prefer to promote pure federative softwares.

If so, it's a pitty because I really like LBRY but it a fair point in the end.

**Comment by mcdviii** [comment]
If that's their position, they should state it. Because they're choosing to be cagey about their motives, we can only assume it's due to selfish political ideology. LBRY may not be "right wing", but they see it as such--that's all that matters.

Reaching to make an excuse for them is a waste of all of our time, honestly. They're not interested in the implementation--the end.

On September 22, 2022 8:37:41 AM CDT, Bruno ***@***.***> wrote:
>Honest question:
>
>What political reasons makes them be against LBRY/Odysee though?
>
>It is just an decentralized alternative to YT ran by individuals servers connected by the BitTorrent Protocol, the Blockchain is simply for indexing trackers.
>
>No right/left wing speech are promoted by them, and they have basically the same rules of YouTube.
>
>=========
>
>My guessing: 
>
>Maybe it's not political though, but tech-idealism? 
>Since they have PeerTube, that is basically the same of Odysee but part of Fediverse my guessing would be that the developers don't actually like blockchain stuff and prefer to promote pure federative softwares.
>
>If so, it's a pitty because I really like LBRY but it a fair point in the end.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/TeamNewPipe/NewPipeExtractor/issues/434#issuecomment-1255038576
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by DraconicNEO** [comment]
> Honest question:
> 
> What political reasons makes them be against LBRY/Odysee though?
> 
> It is just an decentralized alternative to YT ran by individuals servers connected by the BitTorrent Protocol, the Blockchain is simply for indexing trackers.
> 
> No right/left wing speech are promoted by them, and they have basically the same rules of YouTube.
> 
> =========
> 
> My guessing:
> 
> Maybe it's not political though, but tech-idealism? Since they have PeerTube, that is basically the same of Odysee but part of Fediverse my guessing would be that the developers don't actually like blockchain stuff and prefer to promote pure federative softwares.
> 
> If so, it's a pitty because I really like LBRY but it a fair point in the end.

I'm going to say that was probably their whole reason of rejecting platforms in the first place. It wouldn't matter what content is on those platforms it would be how those platforms are set up.

**Comment by mrusme** [comment]
I keep reading these comments and I'm wondering: Are there any sources to back this up? Did the developers of NewPipe **actually** say that they won't implement service X or Y because of ideological reasons?

Unless that's the case I'd give them the benefit of the doubt and probably rather take [this](https://github.com/TeamNewPipe/NewPipe/issues/6645#issuecomment-1016863552) comment as the primary reason for not adding further services.

Long story short, it's OSS. If someone with a bit of time (and necessary knowledge) on their hands feels like building a **solid** LBRY implement and opening a PR I can hardly imagine the NewPipe people acting like *nah thanks, we're good*.

**Comment by mcdviii** [comment]
I look at my hidden question above speaking to this exact issue & I see my answer.

I will agree that debating this is a fucking waste of time--and the topic should be dropped until results are produced (somewhere), however.

On September 22, 2022 5:11:59 PM CDT, "◤◢◤◢◤◢◤◢" ***@***.***> wrote:
>I keep reading these comments and I'm wondering: Are there any sources to back this up? Did the developers of NewPipe **actually** say that they won't implement service X or Y because of ideological reasons?
>
>Unless that's the case I'd give them the benefit of the doubt and probably rather take [this](https://github.com/TeamNewPipe/NewPipe/issues/6645#issuecomment-1016863552) comment as the primary reason for not adding further services.
>
>Long story short, it's OSS. If someone with a bit of time (and necessary knowledge) on their hands feels like building a **solid** LBRY implement and opening a PR I can hardly imagine the NewPipe people acting like *nah thanks, we're good*.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/TeamNewPipe/NewPipeExtractor/issues/434#issuecomment-1255608225
>You are receiving this because you commented.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by strider72** [comment]
> Did the developers of NewPipe **actually** say that they won't implement service X or Y because of ideological reasons?

Yes they did, here: https://github.com/TeamNewPipe/NewPipeExtractor/issues/434#issuecomment-1255038576

They went so far as to argue that you saying something they disagree with is a *violation of their rights*. Pure politics.

**Comment by opusforlife2** [comment]
Right. Lots of speculation, assumptions and accusations in the comments. And some whining, but that's par for the course.

Firstly, there have been no comments by any team member either accepting or rejecting the service. Nor has there ever been a PR adding support that could be accepted or rejected. So any assumptions or accusations based on that are wrong. Some voices of reason questioned this, so thank you for that. Your comments were still marked off-topic because that's just the case on Github. Feel free to go off-topic galore in our IRC channel.

Secondly, just like Bravenewpipe supports 2 additional services that Newpipe doesn't have, you can go all out and add whatever services you like to your own forks. Because FOSS. Or, you could _politely_ request the maintainers of a fork (there are at least 2) to add support. Because FOSS. But complaining about the supposed "political views" of team members won't suddenly make them 'see the light' or the 'error of their ways' and get them to drop everything to spend hours/days/weeks to add support for [insert your personal bandwagon service here] for free. Because FOSS. So there is no point in doing that, in this repo or in any FOSS project.

Finally, Odysee unfortunately caters to misinformation: https://www.reuters.com/investigates/special-report/usa-media-misinformation/

If you genuinely think that [insert your favourite content creator here] is not part of the misinformation camp and doesn't deserve to get blocked out merely because they use LBRY, consider persuading them to try Peertube. Not only will it potentially increase their viewers, but it would help them avoid the stigma associated with LBRY. And of course, you can add their instance, or one their chosen instance federates to, to Newpipe. (Just to be clear, this is not about superior tech, merely about exploring a platform that could help such creators out.)

And now, some food for thought: https://en.wikipedia.org/wiki/Paradox_of_tolerance

Service rejected. If, in the future, it turns out to be the case that a vast majority of LBRY content creators are genuine and the misinformation camp has been reduced to a tiny minority, we could revisit this service request.
